emmet-core/emmet/core/phonon.py

```diff
@@ -1,131 +1,816 @@
+            {f"{col_prefix}{k}": [v] for k, v in config.items()}
+        )
+
+    @classmethod
+    @requires(pa is not None, "`pip install pyarrow` to use this functionality.")
+    def from_arrow(cls, table: ArrowTable, col_prefix: str | None = None) -> Self:
+        """Create a PhononBS from an arrow table."""
+        col_prefix = col_prefix or ""
+        config: dict[str, Any] = {}
+        for k in (
+            "structure",
+            "has_nac",
+            "qpoints",
+            "frequencies",
+            "reciprocal_lattice",
+            "eigendisplacements_real",
+            "qpoint_labels",
+        ):
+            _k = f"{col_prefix}{k}"
+            if _k not in table.column_names:
+                continue
+            v = table[_k].to_pylist()[0]
+            if k == "structure":
+                config[k] = Structure.from_dict(json.loads(v))
+            elif k in ("qpoints", "frequencies", "reciprocal_lattice"):
+                config[k] = np.array(v)
+            elif k == "eigendisplacements_real":
+                config["eigendisplacements"] = (
+                    table[f"{col_prefix}eigendisplacements_real"].to_numpy()[0]
+                    + 1.0j * table[f"{col_prefix}eigendisplacements_imag"].to_numpy()[0]
+                )
+            elif k == "qpoint_labels":
+                config["labels_dict"] = dict(
+                    zip(v, table[f"{col_prefix}qpoint_labelled_points"].to_pylist()[0])
+                )
+            else:
+                config[k] = v
+        return cls(**config)
+
+    @classmethod
+    def from_phonopy(cls, phonon_bandstructure_file: str | Path):
+        """Create a PhononBS from phonopy .yaml output."""
+        with zopen(phonon_bandstructure_file, "rt") as f:
+            phonopy_bandstructure = yaml.safe_load(f.read())
+
+        phonopy_bandstructure.update(
+            qpoints=[entry["q-position"] for entry in phonopy_bandstructure["phonon"]],
+            frequencies=[
+                [branch["frequency"] for branch in entry["band"]]
+                for entry in phonopy_bandstructure["phonon"]
+            ],
+        )
+        return cls(**phonopy_bandstructure)
+
+
+class SumRuleChecks(BaseModel):
+    """Container class for defining sum rule checks."""
+
+    asr: float | None = Field(
+        None, description="The violation of the acoustic sum rule."
+    )
+    cnsr: float | None = Field(
+        None, description="The violation of the charge neutral sum rule."
+    )
+
+
+class PhononBSDOSTask(StructureMetadata):
+    """Phonon band structures and density of states data."""
+
+    identifier: str | None = Field(
+        None, description="The identifier of this phonon analysis task."
+    )
+
+    phonon_method: PhononMethod | None = Field(
+        None, description="The method used to calculate phonon properties."
     )
 
-    ph_bs: Optional[PhononBandStructureSymmLine] = Field(
+    phonon_bandstructure: PhononBS | None = Field(
```

Comment:
Should `phonon_{bandstructure,dos}` be optional since it's outsourced to S3 in production?